### PR TITLE
add mediaSubscriptionId field to CallConnectionProperties class

### PR DESCRIPTION
**Pull request by atazimsft**
Exposing mediaSubscriptionId field in the public CallConnectionProperties class

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallConnectionProperties.java [hunk 1]
```diff
@@ -27,6 +28,7 @@ public final class CallConnectionProperties {
     private final CallConnectionState callConnectionState;
     private final String subject;
     private final URI callbackUri;
+    private String mediaSubscriptionId;
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallConnectionProperties.java:31]: Why is this not final?

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAsyncClientUnitTests.java [hunk 2]
```diff
@@ -85,6 +85,7 @@ public void answerCallWithResponse() {
         assertNotNull(answerCallResult);
         assertEquals(200, answerCallResult.getStatusCode());
         assertNotNull(answerCallResult.getValue());
+        assertEquals("mediaSubscriptionId", answerCallResult.getValue().getCallConnectionProperties().getMediaSubscriptionId());
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAsyncClientUnitTests.java:88]: are all the tests returning a Media_Subscription_Id regardless of the mediaStreamingConfiguration was provided or not?

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationClientUnitTests.java [hunk 3]
```diff
@@ -78,7 +78,7 @@ public void answerCallWithResponse() {
         CallAutomationClient callAutomationClient = getCallAutomationClient(new ArrayList<>(
             Collections.singletonList(
                 new SimpleEntry<>(generateCallProperties(CALL_CONNECTION_ID, CALL_SERVER_CALL_ID,
-                    CALL_CALLER_ID, CALL_TARGET_ID, CALL_CONNECTION_STATE, CALL_SUBJECT, CALL_CALLBACK_URL), 200)
+                    CALL_CALLER_ID, CALL_TARGET_ID, CALL_CONNECTION_STATE, CALL_SUBJECT, CALL_CALLBACK_URL, MEDIA_SUBSCRIPTION_ID), 200)
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationClientUnitTests.java:81]: Here the media subscription id is not even verified?

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAsyncClientUnitTests.java [hunk 4]
```diff
@@ -85,7 +86,7 @@ public void answerCallWithResponse() {
         assertNotNull(answerCallResult);
         assertEquals(200, answerCallResult.getStatusCode());
         assertNotNull(answerCallResult.getValue());
-        assertEquals("mediaSubscriptionId", answerCallResult.getValue().getCallConnectionProperties().getMediaSubscriptionId());
+        assertEquals("mediaSubscriptionId", null);
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallAutomationAsyncClientUnitTests.java:89]: you are checking for null, but your `answerCallWithResponse` seems to be receiving MediaStreamingConfiguration. that doesn't seem to be accurate.

**Review by cochi2 — APPROVED** [review]

**Review by juntuchen-msft — APPROVED** [review]

**Pull request merged**